packages/uploadthing/src/internal/types.ts

```
@@ -230,20 +226,16 @@
 };
 
-export type inferEndpointInput<TUploader extends Uploader<any>> =
-  TUploader["_def"]["_input"]["in"] extends UnsetMarker
-    ? undefined
-    : TUploader["_def"]["_input"]["in"];
+export type inferEndpointInput<TUploader extends AnyUploader> =
+  TUploader["$types"]["input"];
 
 export type inferEndpointOutput<TUploader extends AnyUploader> =
-  TUploader["_def"]["_output"] extends UnsetMarker | void | undefined
+  TUploader["$types"]["output"] extends UnsetMarker | void | undefined
```

Biome

[error] void is confusing inside a union type.

Unsafe fix: Use undefined instead.


(lint/suspicious/noConfusingVoidType)